pwnagotchi/plugins/default/clock.py

```diff
@@ -0,0 +1,147 @@
+
+import pwnagotchi.plugins as plugins
+import pwnagotchi.ui.fonts as fonts
+from pwnagotchi.ui.components import LabeledValue
+from pwnagotchi.ui.view import BLACK
+from socket import socket, AF_INET, SOCK_DGRAM
+import struct
+
+
+class Clock(plugins.Plugin):
+    __author__ = "benleb"
+    __version__ = "1.0.0"
+    __license__ = "GPL3"
+    __plugin__ = "clock"
+    __description__ = "Shows the time, sometimes even the current one."
+
+    def __init__(self):
+        self.synced = False
+
+    def timedatectl_state(self):
+        try:
+            state = subprocess.run(
+                ["timedatectl", "status"],
+                capture_output=True, check=True, text=True, timeout=5
+            )
+
+            matches = re.search(r".*System clock synchronized: (\w*)\W", state.stdout, re.MULTILINE)
+            if matches and matches.group(1) == "yes":
+                logging.info(f"[{self.__plugin__}]: system clock is in sync")
+                self.synced = True
+
+            return self.synced
+
+        except TimeoutExpired as error:
+            logging.error(f"[{self.__plugin__}] timeout: {error}")
+        except CalledProcessError as error:
+            logging.error(f"[{self.__plugin__}] error: {error}")
+
+    def on_loaded(self):
+        logging.info(f"[{self.__plugin__}]: plugin loaded")
+
+    def on_ready(self, agent):
+        self.timedatectl_state()
+        # try:
+        #     state = subprocess.run(
+        #         ["timedatectl", "status"],
+        #         capture_output=True, check=True, text=True, timeout=5
+        #     )
+
+        #     matches = re.search(r".*System clock synchronized: (\w*)\W", state.stdout, re.MULTILINE)
+        #     if matches and matches.group(1) == "yes":
+        #         logging.info(f"[{self.__plugin__}]: system clock is synchronized")
+        #         self.synced = True
+
+        # except TimeoutError as error:
+        #     logging.error(f"[{self.__plugin__}] timeout: {error}")
+        # except CalledProcessError as error:
+        #     logging.error(f"[{self.__plugin__}] error: {error}")
+
+    def on_ui_setup(self, ui):
+        if ui.is_waveshare_v2():
+            pos = (208, 67)
+        elif ui.is_inky():
+            # guessed values, add tested ones if you can
+            pos = (112, 30)
+        else:
+            # guessed values, add tested ones if you can
+            pos = (127, 51)
+
+        label_spacing = 0
+
+        ui.add_element(
+            "clock",
+            LabeledValue(
+                color=BLACK,
+                label="",
+                value="-:-:-",
+                position=pos,
+                label_font=fonts.Small,
+                text_font=fonts.Small,
+                label_spacing=label_spacing,
+            ),
+        )
+
+        if self.synced:
+            ui.set("clock", f"{datetime.now().time().isoformat(timespec='seconds')} ")
+        elif self.query_ntp():
+            ui.set("clock", f"{self.query_ntp()} ")
```

Comment:
Since you're firing off a udp connection to the ntp server and waiting for a response every query_ntp() is called you really don't want to wait for a response twice. Save the result, then set the clock appropriately if you get a good result rather than querying the server twice and throwing away the first result.